tools/clang/lib/Sema/SemaStmt.cpp

```diff
@@ -2709,6 +2701,38 @@ VarDecl *Sema::getCopyElisionCandidate(QualType ReturnType,
   if (!VD)
     return nullptr;
 
+  // HLSL Change Begins: NRVO unsafe for a variety of cases in HLSL
+  // - vectors/matrix with bool component types
+  // - attributes not captured to QualType, such as precise and globallycoherent
+  // NRVO is also unnecessary for HLSL correctness or performance.
```

Comment:
Comment seems out-of-date now, since it's required to avoid extra memcpy and memcpy elimination.  I wish there was a way to fix that part (handle extra memcpy) though.

Reply:
SROA should make the memcpy go away, but I suspect the version we're using in DXC doesn't match the case.